components/script/dom/htmltablecellelement.rs

```diff
@@ -123,6 +128,20 @@ impl HTMLTableCellElementLayoutHelpers for LayoutJS<HTMLTableCellElement> {
                 .unwrap_or(LengthOrPercentageOrAuto::Auto)
         }
     }
+
+    fn get_inherited_cellpadding(&self) -> Option<u32>
+    {
+        unsafe {
+            // Follow chain: Cell -> Row -> Section -> Table
+            self.upcast::<Node>().parent_node_ref()
```

Comment:
I believe tables are always constructed with this structure in Servo. If the input document's table isn't well formed, these elements are still inserted into the tree. This patch is OK in the anonymous row example, and where there are nested table rows.

Looking into other browsers, Firefox takes a similar route, with an additional case for where there is no HTMLTableSectionElement equivalent.
https://dxr.mozilla.org/mozilla-central/source/dom/html/HTMLTableCellElement.cpp#119
https://dxr.mozilla.org/mozilla-central/source/dom/html/HTMLTableCellElement.cpp#47
I'm not sure we need that here - I think there is always a tbody between the table and row.
https://github.com/servo/servo/blob/81a382cb00134450f3723caf682bb5ca8cca317f/components/script/dom/htmltableelement.rs#L306

Chrome searches up the tree until it finds a table element. This approach does need less code.
https://cs.chromium.org/chromium/src/third_party/WebKit/Source/core/html/HTMLTablePartElement.cpp?l=110&rcl=6ce68a3fb3e459bb4e9676a508a9260d4e67bd10
The short loop might be faster on average. I'm not sure how I'd feel about the possibility to search all the way up the tree though.

The spec isn't particularly helpful here https://www.w3.org/TR/html4/struct/tables.html#adef-cellpadding